src/net/network_manager.cpp

```diff
@@ -273,6 +274,9 @@ void ClientNetworkManager::update() {
             assert(player);
             player->c_update_stats(update);
 
+            if (client_player && update.id() == client_player->get_id())
+                events::player_coins_update_event(update.coins());
```

Comment:
`NetworkManager` should absolutely not be concerned with the semantics of a stats update, and should not even know about coins. Move this to `Player::c_update_stats`.